chatkit-core/src/main/kotlin/com/pusher/chatkit/state/Reducers.kt

```diff
@@ -0,0 +1,16 @@
+package com.pusher.chatkit.state
+
+import org.reduxkotlin.ReducerForActionType
+import org.reduxkotlin.reducerForActionType
+
+/**
+ * Convenience method to create a reducer with operates only on the ChatState section of the State
+ * hierarchy.
+ */
+internal inline fun <reified TAction> chatReducerForActionType(
+    crossinline reducer: ReducerForActionType<ChatState, TAction>
+) = reducerForActionType<State, TAction> { state, action ->
+    state.with(
+        chatState = reducer(state.chatState, action)
```

Comment:
Nice I like it! It's good that it's the code written once. Moreover it's adherent to the redux holy scripture and those reducers even earned a name there: slice reducers ([reference](https://redux.js.org/recipes/structuring-reducers/splitting-reducer-logic)) 😉

Reply:
Honestly, the `reducerForActionType` seems a bit verbose. We could have `chatReducer` and `stateReducer` where the State type var is pre-bound, because it's always the same and it's always going to be for an action type (in our code) so that's a bit redundant in the function name.

Reply:
I've pushed a rename to `chatStateReducer` to reduce verbosity.